Panic on 0.10.3 when init with consul

### Terraform Version
Terraform v0.10.3

### Terraform Configuration Files
```hcl
terraform {
  backend "consul" {
    address = "localhost:32847"
    path    = "/kv/tfstate"
  }
}
```

### Debug Output
see crash output 

### Crash Output

```
2017/08/31 09:08:16 [INFO] Terraform version: 0.10.3  1511d447e78d8106787875265f96d3a3a03005d5
2017/08/31 09:08:16 [INFO] Go runtime version: go1.9
2017/08/31 09:08:16 [INFO] CLI args: []string{"/usr/local/bin/terraform", "init", "init/"}
2017/08/31 09:08:16 [DEBUG] Attempting to open CLI config file: /home/jeroen/.terraformrc
2017/08/31 09:08:16 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2017/08/31 09:08:16 [INFO] CLI command args: []string{"init", "init/"}
2017/08/31 09:08:16 [DEBUG] command: loading backend config file: /home/jeroen/tmp/tf/x/init

2017/08/31 09:08:16 [DEBUG] command: no data state file found for backend config
2017/08/31 09:08:16 [DEBUG] New state was assigned lineage "98503445-e06b-4d0d-87e1-45e6c2c36cdc"
Initializing the backend...
2017/08/31 09:08:16 [DEBUG] plugin: waiting for all plugin processes to complete...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xef93af]

goroutine 1 [running]:
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).clientRaw(0xc4202e4e40, 0x21ec627, 0x4, 0x1b03880)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:152 +0x3af
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).configure(0xc4202e4e40, 0x35e5b80, 0xc4202dd230, 0x1b0b900, 0xc4202c5a00)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:116 +0x9f
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).(github.com/hashicorp/terraform/backend/remote-state/consul.configure)-fm(0x35e5b80, 0xc4202dd230, 0x1b0b900, 0xc4202c5a00)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:96 +0x3e
github.com/hashicorp/terraform/helper/schema.(*Backend).Configure(0xc4202e4e20, 0xc4202dcde0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/backend.go:80 +0x186
github.com/hashicorp/terraform/command.(*Meta).backendInitFromConfig(0xc420001800, 0xc4202e4c80, 0x3, 0xc420000180, 0xc42024a000, 0x3)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:1356 +0x1b1
github.com/hashicorp/terraform/command.(*Meta).backend_C_r_s(0xc420001800, 0xc4202e4c80, 0x35e87e0, 0xc420067f80, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:959 +0x8b
github.com/hashicorp/terraform/command.(*Meta).backendFromConfig(0xc420001800, 0xc4201c7ae8, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:384 +0xee6
github.com/hashicorp/terraform/command.(*Meta).Backend(0xc420001800, 0xc4201c7ae8, 0x28, 0xc4200195c0, 0x28, 0xc420019590)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:88 +0x83c
github.com/hashicorp/terraform/command.(*InitCommand).Run(0xc420001800, 0xc42000e0b0, 0x1, 0x1, 0xc4201a1fd0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/init.go:200 +0xd32
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc42026f0e0, 0xc42026f0e0, 0xc4201c7cf8, 0x1)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:242 +0x1eb
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:188 +0xac0
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:95 +0xbb
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:31 +0x38
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xef93af]

goroutine 1 [running]:
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).clientRaw(0xc4202e4e40, 0x21ec627, 0x4, 0x1b03880)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:152 +0x3af
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).configure(0xc4202e4e40, 0x35e5b80, 0xc4202dd230, 0x1b0b900, 0xc4202c5a00)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:116 +0x9f
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).(github.com/hashicorp/terraform/backend/remote-state/consul.configure)-fm(0x35e5b80, 0xc4202dd230, 0x1b0b900, 0xc4202c5a00)
	/opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:96 +0x3e
github.com/hashicorp/terraform/helper/schema.(*Backend).Configure(0xc4202e4e20, 0xc4202dcde0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/helper/schema/backend.go:80 +0x186
github.com/hashicorp/terraform/command.(*Meta).backendInitFromConfig(0xc420001800, 0xc4202e4c80, 0x3, 0xc420000180, 0xc42024a000, 0x3)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:1356 +0x1b1
github.com/hashicorp/terraform/command.(*Meta).backend_C_r_s(0xc420001800, 0xc4202e4c80, 0x35e87e0, 0xc420067f80, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:959 +0x8b
github.com/hashicorp/terraform/command.(*Meta).backendFromConfig(0xc420001800, 0xc4201c7ae8, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:384 +0xee6
github.com/hashicorp/terraform/command.(*Meta).Backend(0xc420001800, 0xc4201c7ae8, 0x28, 0xc4200195c0, 0x28, 0xc420019590)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:88 +0x83c
github.com/hashicorp/terraform/command.(*InitCommand).Run(0xc420001800, 0xc42000e0b0, 0x1, 0x1, 0xc4201a1fd0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/init.go:200 +0xd32
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc42026f0e0, 0xc42026f0e0, 0xc4201c7cf8, 0x1)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:242 +0x1eb
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:188 +0xac0
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:95 +0xbb
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:31 +0x38



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

### Expected Behavior
initialise the consul backend

### Actual Behavior
crash 

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
0. place hcl file in init directory
1. `terraform init ./init`


### Important Factoids
running from the command line
consul is running in a container (hence the weird port)
```
$ nc -v localhost 32847
Connection to localhost 32847 port [tcp/*] succeeded!
```

### References
-

Tried to upgrade to 0.10.3 (from 0.10.2) and also had a crash:

`[root@se-centos7 nomad]# /bin/terraform -v
Terraform v0.10.3

[root@se-centos7 nomad]# terraform plan
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xef93af]

goroutine 1 [running]:
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).clientRaw(0xc42036a180, 0x21ec627, 0x4, 0x1b03880)
        /opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:152 +0x3af
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).configure(0xc42036a180, 0x35e5b80, 0xc42035b320, 0x1b0b900, 0xc42036ca60)
        /opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:116 +0x9f
github.com/hashicorp/terraform/backend/remote-state/consul.(*Backend).(github.com/hashicorp/terraform/backend/remote-state/consul.configure)-fm(0x35e5b80, 0xc42035b320, 0x1b0b900, 0xc42036ca60)
        /opt/gopath/src/github.com/hashicorp/terraform/backend/remote-state/consul/backend.go:96 +0x3e
github.com/hashicorp/terraform/helper/schema.(*Backend).Configure(0xc42036a160, 0xc42035af00, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/helper/schema/backend.go:80 +0x186
github.com/hashicorp/terraform/command.(*Meta).backend_C_r_S_unchanged(0xc4202ee160, 0xc4202f9b80, 0x35e87e0, 0xc4203500c0, 0x1c, 0xc4201b17a0, 0x3, 0xc4201b1850)
        /opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:1188 +0x1bd
github.com/hashicorp/terraform/command.(*Meta).backendFromConfig(0xc4202ee160, 0xc4201b1ae8, 0x0, 0x0, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:398 +0xdcc
github.com/hashicorp/terraform/command.(*Meta).Backend(0xc4202ee160, 0xc4201b1ae8, 0x18, 0xc420241f40, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/meta_backend.go:88 +0x83c
github.com/hashicorp/terraform/command.(*PlanCommand).Run(0xc4202ee160, 0x0, 0x0, 0x0, 0xc4202451f0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/plan.go:86 +0x6b9
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc420247560, 0xc420247560, 0xc4201b1cf8, 0x1)
        /opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:242 +0x1eb
main.wrappedMain(0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:188 +0xac0
main.realMain(0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:95 +0xbb
main.main()
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:31 +0x38



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
`

--- a/backend/remote-state/consul/backend.go
+++ b/backend/remote-state/consul/backend.go
@@ -2,7 +2,6 @@ package consul
 
 import (
 	"context"
-	"net/http"
 	"strings"
 
 	consulapi "github.com/hashicorp/consul/api"
@@ -149,7 +148,7 @@ func (b *Backend) clientRaw() (*consulapi.Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	config.HttpClient.Transport.(*http.Transport).TLSClientConfig = cc
+	config.Transport.TLSClientConfig = cc
 
 	if v, ok := data.GetOk("http_auth"); ok && v.(string) != "" {
 		auth := v.(string)

